server/src/testFixtures/java/org/elasticsearch/repositories/azure/AzureHttpHandler.java

```diff
@@ -177,13 +186,22 @@ public void handle(final HttpExchange exchange) throws IOException {
                     list.append("<Blob><Name>").append(blobPath).append("</Name>");
                     list.append("<Properties><Content-Length>").append(blob.getValue().length()).append(
                         "</Content-Length>");
+                    List<String> dateHeader = exchange.getRequestHeaders().get("X-ms-date");
+                    if (dateHeader != null) {
+                        // COPY/OpenDAL requires this header.
+                        // CREATE REPOSITORY/dedicated client fails on encountering header in such format.
+                        // TODO: Migrate CREATE REPOSITORY on Azure to use OpenDAL.
+                        list.append("<Last-Modified>").append(dateHeader.get(0)).append("</Last-Modified>");
+                    }
```

Comment:
Some settings are common but Repository uses dedicated library and it doesn't play well with OpenDAL. I added todo to maybe use OpenDAL in es-azure plugin as well but for now I added dedicated `AzureBlobStorageSettings`.